### PR TITLE
edge-21.1.4

### PR DESCRIPTION
**Pull request by adleong**
This edge release continues to polish the Linkerd extension model and improves
the robustness of the opaque transport.

* Improved the consistency of behavior of the `check` commands between
  Linkerd extensions
* Fixed an issue where Linkerd extension commands could be run before the
  extension was fully installed
* Renamed some extension Helm charts for consistency:
  * jaeger -> linkerd-jaeger
  * linkerd2-multicluster -> linkerd-multicluster
  * linkerd2-multicluster-link -> linkerd-multicluster-link
* Fixed an issue that could cause the inbound proxy to fail meshed HTTP/1
  requests from older proxies (from the stable-2.8.x vintage)
* Changed opaque-port transport to be advertised via ALPN so that new proxies
  will not initiate opaque-transport connections to proxies from prior
  edge releases
* Added inbound proxy transport metrics with `tls="passhtru"` when forwarding
  non-mesh TLS connections
* Thanks to @hs0210 for adding new unit tests!

Signed-off-by: Alex Leong <alex@buoyant.io>

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,6 +1,28 @@
...
+* Fixed an issue where Linkerd extension commands could be run before the
+  extension was fully installed
+* Renamed some extension Helm charts for consistency:
+  * jaeger -> linkerd-jaeger
+  * linkerd2-multicluster -> linkerd-multicluster
+  * linkerd2-multicluster-link -> linkerd-multicluster-link
+* Fixed an issue that could cause the inbound proxy to fail meshed HTTP/1
+  requests from older proxies (from the stable-2.8.x vintage)
+* Changed opaque-port transport to be advertised via ALPN so that new proxies
+  will not initiate initiate opaque-transport connections to proxies from prior
```
Comment (dadjeibaah) [inline CHANGES.md:20]: rm initiate

File: CHANGES.md [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+* Improved the consistency of behavior of the `check` commands between
+  Linkerd extensions
+* Fixed an issue where Linkerd extension commands could be run before the
+  extension was fully installed
+* Renamed some extension Helm charts for consistency:
+  * jaeger -> linkerd-jaeger
+  * linkerd2-multicluster -> linkerd-multicluster
+  * linkerd2-multicluster-link -> linkerd-multicluster-link
+* Fixed an issue that could cause the inbound proxy to fail meshed HTTP/1
+  requests from older proxies (from the stable-2.8.x vintage)
```
Comment (dadjeibaah) [inline CHANGES.md:18]: I'm not quite sure I understand how `vintage` is being used here. Is it referring to older proxies from all stable-2.8.x releases?

**Review by dadjeibaah — COMMENTED** [review]
Left a couple of comments otherwise its looking pretty good!

**Review by kleimkuhler — APPROVED** [review]
🆗!

File: CHANGES.md (hunk 2 above)
Comment (olix0r) [reply CHANGES.md:18]: yeah, that's what i meant there.

**Review by olix0r — APPROVED** [review]

**Pull request merged**